scripts/modules/core/interfaces.js

```
@@ -0,0 +1,207 @@
+ */
+export function validateInterface(obj, interfaceDefinition, name = 'object') {
+	const missing = [];
+
+	for (const [key, expectedType] of Object.entries(interfaceDefinition)) {
+		if (!(key in obj)) {
+			missing.push(key);
+		} else if (expectedType === Function && typeof obj[key] !== 'function') {
+			missing.push(`${key} (expected function, got ${typeof obj[key]})`);
+		} else if (
+			expectedType !== Function &&
+			typeof obj[key] !== expectedType.name.toLowerCase()
```

Biome (1.9.4)

[error] Invalid `typeof` comparison value: this expression is not a string literal

not a string literal

(lint/suspicious/useValidTypeof)